pkg/oc/cli/admin/mustgather/mustgather.go

```diff
@@ -243,6 +273,16 @@ func (o *MustGatherOptions) waitForPodRunning(pod *corev1.Pod) error {
 	return nil
 }
 
+func (o *MustGatherOptions) waitForGatherContainerRunning(pod *corev1.Pod) error {
+	return wait.PollImmediate(time.Second, 10*time.Minute, func() (bool, error) {
+		var err error
+		if pod, err = o.Client.CoreV1().Pods(pod.Namespace).Get(pod.Name, metav1.GetOptions{}); err != nil {
+			return false, nil
```

Comment:
return false only on NotFound, otherwise return the error

Reply:
reworked using `retry.RetryOnConnectionErrors()`